railties/lib/rails/engine/configuration.rb

```diff
@@ -89,7 +89,8 @@ def autoload_paths
 
       def webpacker_path
         if File.file?("#{Rails.root}/config/webpacker.yml")
-          YAML.load_file("#{Rails.root}/config/webpacker.yml")[Rails.env]["source_path"]&.gsub("app/", "")
+          webpacker_env = YAML.load_file("#{Rails.root}/config/webpacker.yml")[Rails.env].blank? ? "production" : Rails.env
```

Comment:
I am concerned that this very depends on Webpacker's specification. Is it difficult to use Webpacker's method? I assume the following code. 

```diff 
       def webpacker_path
-        if File.file?("#{Rails.root}/config/webpacker.yml")
-          YAML.load_file("#{Rails.root}/config/webpacker.yml")[Rails.env]["source_path"]&.gsub("app/", "")
-        else
+        @webpacker_path ||= begin
+          require "webpacker"
+
+          Webpacker.config.source_path.relative_path_from(Rails.root.join("app")).to_s
+        rescue LoadError
           "javascript"
         end 
```

Reply:
i will give this a shot, seems like a better idea to use this method

Reply:
it is working, however tests are failing since the webpacker method is producing output that isn't being expected prior to this change

Reply:
@andrewkress : I am not completely sure but after looking at the failing build it looks like the assertion is failing because of the warning messages `RAILS_ENV=test environment is not defined in config/webpacker.yml, falling back to production environment`

You can refer to the following comment. They solved the issue by adding `system({"SKIP_REQUIRE_WEBPACKER" => "true"}` this. Hope that helps 😊 

https://github.com/rails/rails/pull/34009#discussion_r221787457

Reply:
@abhaynikam the changes discussed in that commit are where some of the tests are failing.  It appears that calling these methods is triggering it to come back.

Reply:
@andrewkress : I think the reason for showing the warning is because app is requiring webpacker even when it does not have javascript. Example when using `skip-javascript`.

In the test, `run_generator` is used which by default skip the webpack here(https://github.com/rails/rails/blob/master/railties/lib/rails/generators/testing/behaviour.rb#L71) due to which webpacker config is not generated and the warning is printed on the console. If you do following changes, the test case will pass(It passed on my local).
```
--- a/railties/test/generators/app_generator_test.rb
+++ b/railties/test/generators/app_generator_test.rb
@@ -217,7 +217,7 @@ def test_new_application_doesnt_need_defaults

   def test_new_application_load_defaults
     app_root = File.join(destination_root, "myfirstapp")
-    run_generator [app_root]
+    run_generator [app_root, "--no-skip-webpack-install"]
```

IMO, we should not require webpacker unless `webpacker.yml` file is present. I would suggest doing following in the 
```
def webpacker_path
  return "javascript" unless File.file?("#{Rails.root}/config/webpacker.yml")
  @webpacker_path ||= begin
    require "webpacker"

    Webpacker.config.source_path.relative_path_from(Rails.root.join("app")).to_s
  rescue LoadError, Errno::ENOENT, RuntimeError, NoMethodError
    "javascript"
  end
end
```

Sorry about my first comment. I hadn't checked which cases were failing exactly just commented it directly. 😊

Reply:
Also, with the change, I suggested above(the second one), both cases passed on my local.

Reply:
yea, the failing one's passed locally for me now.  Thanks!